File: src/com/leetcode/Easy/IntersectionOfTwoArrays.java
package com.leetcode.Easy;

import java.util.*;

// 349. Intersection of Two Arrays
public class IntersectionOfTwoArrays {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();

        for (int i = 0; i < nums1.length; i++) {
            hashMap.put(nums1[i], 1);
        }

        for (int i = 0; i < nums2.length; i++) {
            if (hashMap.containsKey(nums2[i])) {
                int value = hashMap.get(nums2[i]);
                hashMap.put(nums2[i], ++value);
            }
        }

        List<Integer> list = new ArrayList<>();
        for (Map.Entry<Integer, Integer> entry : hashMap.entrySet()) {
            if (entry.getValue() > 1) {
                list.add(entry.getKey());
            }
        }
        int[] arr = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            arr[i] = list.get(i);
        }
        return arr;
    }
}

File: JavaScript/Easy/IntersectionOfTwoArrays.js
/**
 * 349. Intersection of Two Arrays
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
const intersection = function (nums1, nums2) {
  let set = new Set(nums1);
  let res = new Set();

  for (let num of nums2) {
    if (set.has(num)) {
      res.add(num);
    }
  }

  return Array.from(res);
};